### PR TITLE
Integrate Generator and Detector classes into PoseDetector code

### PR DESCRIPTION
**Pull request by michelllepan**

### CODE REVIEW

File: batch_test.py [hunk 1]
```diff
@@ -2,8 +2,14 @@
 import csv
 import cv2
 import argparse
-from cv import PoseDetector
-from pathlib import Path
+from pose_detector import PoseDetector
+# from depth_data_generator import DepthDataGenerator
+# from depth_live_generator import DepthLiveGenerator
+from image_generator import ImageGenerator
```
Comment (AG6GR) [inline batch_test.py:6]: Most of these generators aren't used, so we don't need these imports.

File: pose_detector.py [hunk 2]
```diff
@@ -12,13 +7,20 @@
 import os
 from math import sin, cos
 from operator import add
+# from depth_data_generator import DepthDataGenerator
```
Comment (AG6GR) [inline pose_detector.py]: Detector classes should not import any generator classes

File: pose_detector.py [hunk 3]
```diff
@@ -12,13 +7,20 @@
...
+# from depth_live_generator import DepthLiveGenerator
+from image_generator import ImageGenerator
+from video_file_generator import VideoFileGenerator
+from video_live_generator import VideoLiveGenerator
+from loading_bay_detector import LoadingBayDetector
+from power_port_detector import PowerPortDetector
 
 
 # finds rotation and translation of vision targets
 class PoseDetector:
```
Comment (AG6GR) [inline pose_detector.py]: I just realized that naming the the situational awareness code *Detector is a bit confusing. Not a huge issue though.

Comment (michelllepan) [reply pose_detector.py]: I wasn't sure what to name it, would PoseCalculator work?

**Review by pguduguntla — APPROVED** [review]
Looks good – No comments other than what Sunny posted.

**Pull request merged**